internal/mutator/pod/agent/injector.go

```diff
@@ -367,6 +360,28 @@ func (i *injector) shouldInjectContainer(ctx context.Context, pod *corev1.Pod, n
 	return matchPolicies(pod, ns, i.config.Policies), nil
 }
 
+func (i *injector) canInjectContainer(pod *corev1.Pod, containerToInject corev1.Container) error {
+	// Checking if there is any overlapping with the volumes we want to mount and the volumes already present.
+	duplicateVolumeName := getDuplicateVolumeName(toEmptyDirVolumes(containerToInject.VolumeMounts), pod.Spec.Volumes)
+	if duplicateVolumeName != "" {
+		return fmt.Errorf("one of the volumes of the container to inject is already present: %s", duplicateVolumeName)
+	}
+
+	return nil
+}
+
+func getDuplicateVolumeName(v1 []corev1.Volume, v2 []corev1.Volume) string {
```

Comment:
I do not like particularly this function name, however given that I would like to return the name, I can't think to something better

Reply:
I'm not sure, but how about:
```go
func (i *injector) canInjectContainer(pod *corev1.Pod, containerToInject corev1.Container) error {
  volumes := pod.Spec.Volumes
  duplicateVolumeNames := getDuplicateVolumeNames(append(volumes, toEmptyDirVolumes(containerToInject.VolumeMounts)...))

  // Checking if there is any overlapping with the volumes we want to mount and the volumes already present.
  if len(duplicateVolumeNames) > 0 {
    return fmt.Errorf("injecting agent would produce duplicate Pod volumes: %s",
      strings.Join(duplicateVolumeNames, ","))
  }

  return nil
}

func getDuplicateVolumeNames(volumes []corev1.Volume) []string {
  duplicates := []string{}
  unique := map[string]struct{}{}

  for _, v := range volumes {
    if _, ok := unique[v.Name]; ok {
      duplicates = append(duplicates, v.Name)
    }

    unique[v.Name] = struct{}{}
  }

  return duplicates
}

```